### PR TITLE
AE-508 update forta version

### PR DESCRIPTION
**Pull request by rossAE**
Renamed files to conform with forta-agent 0.0.18 and wrote tests for anomalous-value-js and transfer-ownership.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
 # forta-examples
 Malicious Address Detection
+JS examples have been updated to forta agent v0.0.18
```
Comment (arbitrary-kdawg) [inline README.md]: This looks like a changelog entry.  I would revert this change and instead, to make the README file useful, we should list all the directory names with a short description of each agent.

File: anomalous-value-js/src/agent.js [hunk 2]
```diff
@@ -6,8 +6,9 @@ const { Finding, FindingSeverity, FindingType, getJsonRpcUrl } = require("forta-
 const contractAddresses = {};
 const provider = new ethers.providers.getDefaultProvider(getJsonRpcUrl());
 
-function provideHandleTransaction(provider) {
+function provideHandleTransaction(contractAddresses, provider) {
   return async function handleTransaction(txEvent) {
+    //console.log(contractAddresses[txEvent.to]);
```
Comment (arbitrary-kdawg) [inline anomalous-value-js/src/agent.js]: Did you intend to keep this debug code?

File: anomalous-value-js/src/agent.spec.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+const { provideHandleTransaction } = require("./agent");
+const provider = new ethers.providers.getDefaultProvider(getJsonRpcUrl());
+
+const oneGwei = new BigNumber(ethers.utils.parseUnits("1", "gwei").toString());
+
+function gwei(numGwei) {
+  return oneGwei.times(numGwei);
+}
+
+describe("anomolus value agent", () => {
```
Comment (arbitrary-kdawg) [inline anomalous-value-js/src/agent.spec.js]: typo: "anomolus"

File: anomalous-value-js/src/agent.spec.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      value: value
+    };
+    const receipt = {};
+    const traces = [];
+    const addresses = [];
+    const block = {};
+    return new TransactionEvent(type, network, transaction, receipt, traces, addresses, block);
+  };
+
+  //eventually put some mock rolling statistics on this test address
```
Comment (arbitrary-kdawg) [inline anomalous-value-js/src/agent.spec.js]: should this comment be marked TODO?

File: anomalous-value-js/src/agent.spec.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+      const findings = await handleTransaction(txEvent);
+
+      expect(mockRollingMath.getAverage).toHaveBeenCalledTimes(1);
+      expect(mockRollingMath.getStandardDeviation).toHaveBeenCalledTimes(1);
+      expect(mockRollingMath.getNumElements).toHaveBeenCalledTimes(1);
+      
+      expect(findings).toStrictEqual([
+        Finding.fromObject({
+          name: "High Value",
```
Comment (arbitrary-kdawg) [inline anomalous-value-js/src/agent.spec.js]: "High Value" is relative--this agent detects outliers.  The name "Anomalous High Value Transaction" would be more descriptive.

File: anomalous-value-js/src/agent.spec.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+          description: `Value: ${value}`,
```
Comment (arbitrary-kdawg) [inline anomalous-value-js/src/agent.spec.js]: Also include the contract address in the description
"Contract 0xABCD was sent XYZ Ether"

File: anomalous-value-js/src/agent.js [hunk 7]
```diff
@@ -6,8 +6,9 @@ const { Finding, FindingSeverity, FindingType, getJsonRpcUrl } = require("forta-
 const contractAddresses = {};
 const provider = new ethers.providers.getDefaultProvider(getJsonRpcUrl());
 
-function provideHandleTransaction(provider) {
+function provideHandleTransaction(contractAddresses, provider) {
   return async function handleTransaction(txEvent) {
```
Comment (arbitrary-kdawg) [inline anomalous-value-js/src/agent.js:16]: I think a description of the handler would be useful here--you have to read into the code to figure out that this agent records transaction values for all contracts that it processes and identifies unusually high transaction values.

File: anomalous-gas-js/src/agent.js [hunk 8]
```diff
@@ -1,4 +1,3 @@
-const ethers = require("ethers");
```
Comment (arbitrary-kdawg) [inline anomalous-gas-js/src/agent.js:1]: The code in these examples was never fully linted.  Eslint currently shows 269 issues.  After running with the --fix option, here is what's left:

```
/home/ubuntu/forta-uma-suite/src/examples/anomalous-gas-js/src/agent.js
  3:29  error  Unable to resolve path to module 'rolling-math'                          import/no-unresolved
  8:35  error  'rollingMath' is already declared in the upper scope on line 6 column 7  no-shadow

/home/ubuntu/forta-uma-suite/src/examples/anomalous-value-js/src/agent.js
   9:39  error  A constructor name should not start with a lowercase letter                    new-cap
  11:35  error  'contractAddresses' is already declared in the upper scope on line 8 column 7  no-shadow
  11:54  error  'provider' is already declared in the upper scope on line 9 column 7           no-shadow
  43:51  error  Expected '!==' and instead saw '!='                                            eqeqeq
  45:7   error  Assignment to property of function parameter 'contractAddresses'               no-param-reassign

/home/ubuntu/forta-uma-suite/src/examples/anomalous-value-js/src/agent.spec.js
   3:7   error  'RollingMath' is assigned a value but never used             no-unused-vars
  14:39  error  A constructor name should not start with a lowercase letter  new-cap

/home/ubuntu/forta-uma-suite/src/examples/big-tx-js/src/agent.js
  2:7  error  'BigNumber' is assigned a value but never used  no-unused-vars

/home/ubuntu/forta-uma-suite/src/examples/big-tx-js/src/agent.spec.js
  11:1  error  This line has a length of 109. Maximum allowed is 100  max-len

/home/ubuntu/forta-uma-suite/src/examples/deployment-blacklist-js/src/agent.js
  1:7   error  'ethers' is assigned a value but never used         no-unused-vars
  2:7   error  'BigNumber' is assigned a value but never used      no-unused-vars
  4:42  error  'getJsonRpcUrl' is assigned a value but never used  no-unused-vars

/home/ubuntu/forta-uma-suite/src/examples/deployment-blacklist-js/src/agent.spec.js
  14:1  error  This line has a length of 167. Maximum allowed is 100  max-len

/home/ubuntu/forta-uma-suite/src/examples/function-call-js/src/agent.spec.js
  68:9   error  'to' is assigned a value but never used    no-unused-vars
  68:25  error  'data' is assigned a value but never used  no-unused-vars

/home/ubuntu/forta-uma-suite/src/examples/transfer-ownership/src/agent.js
  15:27  error  Expected '!==' and instead saw '!='  eqeqeq
```

File: anomalous-value-js/src/agent.spec.js (hunk 5 above)
Comment (arbitrary-kdawg) [inline anomalous-value-js/src/agent.spec.js]: "High Value" is relative--this agent detects outliers.  A better name might be "Abnormally High Transaction Value"

File: anomalous-value-js/src/agent.spec.js (hunk 6 above)
Comment (arbitrary-kdawg) [inline anomalous-value-js/src/agent.spec.js]: Might want to include the contract address here as well:
"Transaction with contract 0xABCD had unusually high value of XYZ Ether"

File: transfer-ownership/src/agent.spec.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  
+      it('returns a finding if an event was in the watch list', async () => {
+        const event = EVENTS[1];
+        const txEvent = createTxEvent(event);
+  
+        const findings = await handleTransaction(txEvent);
+  
+        expect(findings).toStrictEqual([
+            Finding.fromObject({
+                name: event.split("(")[0],
```
Comment (arbitrary-kdawg) [inline transfer-ownership/src/agent.spec.js]: Add a comment in the source code that shows an example of what will be printed for 'name' and 'description'

File: transfer-ownership/src/agent.spec.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
+const {
+    TransactionEvent, FindingType, FindingSeverity, Finding,
+  } = require('forta-agent');
+const ethers = require('ethers');
+const { handleTransaction } = require('./agent');
+
+//Take same list of monitored events as used in the agent script
+const ABI = require("./events.json");
```
Comment (arbitrary-kdawg) [inline transfer-ownership/src/agent.spec.js]: Suggest changing ABI to abi and EVENTS to events just for consistency...we've been using all caps only for hardcoded basic types - integers and strings.

File: anomalous-value-js/src/agent.spec.js (hunk 4 above)
Comment (rossAE) [reply anomalous-value-js/src/agent.spec.js]: Changed to "initialize a test address ready for mock RollingMath statistics"

File: anomalous-value-js/src/agent.js (hunk 2 above)
Comment (rossAE) [reply anomalous-value-js/src/agent.js]: Nope, thanks for catching

File: anomalous-value-js/src/agent.spec.js (hunk 3 above)
Comment (rossAE) [reply anomalous-value-js/src/agent.spec.js]: fixed

File: anomalous-value-js/src/agent.spec.js (hunk 5 above)
Comment (rossAE) [reply anomalous-value-js/src/agent.spec.js]: Agreed, fixed

Comment (rossAE) [reply anomalous-value-js/src/agent.spec.js]: Agreed, fixed

File: transfer-ownership/src/agent.spec.js (hunk 10 above)
Comment (rossAE) [reply transfer-ownership/src/agent.spec.js]: Done

File: README.md (hunk 1 above)
Comment (rossAE) [reply README.md]: Done

File: anomalous-value-js/src/agent.spec.js (hunk 6 above)
Comment (rossAE) [reply anomalous-value-js/src/agent.spec.js]: Done

Comment (rossAE) [reply anomalous-value-js/src/agent.spec.js]: Done

File: anomalous-value-js/src/agent.js (hunk 7 above)
Comment (rossAE) [reply anomalous-value-js/src/agent.js:16]: Included the following comment when `contractAddresses` is initialized: "Need to initialize a dictionary whose keys will be contract addresses the agent encounters and whose values are RollingMath objects tracking the ether values of transactions that touch the contract"

File: transfer-ownership/src/agent.spec.js (hunk 9 above)
Comment (rossAE) [reply transfer-ownership/src/agent.spec.js]: Done

File: anomalous-gas-js/src/agent.js (hunk 8 above)
Comment (rossAE) [reply anomalous-gas-js/src/agent.js:1]: Linted!

File: function-call-js/src/agent.js [hunk 11]
```diff
@@ -19,7 +19,7 @@ async function handleTransaction(txEvent) {
       && (txEvent.transaction.data.startsWith(METHOD_ID))) {
     findings.push(
       Finding.fromObject({
-        name: 'USDT Transfer',
+        name: 'AE-FUNCTION-CALLED',
```
Comment (arbitrary-kdawg) [inline function-call-js/src/agent.js]: name is assigned to the alert ID

File: transfer-ownership-js/package.json [hunk 12]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "ownership-transfered-agent",
```
Comment (arbitrary-kdawg) [inline transfer-ownership-js/package.json]: typo

**Review by arbitrary-kdawg — APPROVED** [review]
